src/NServiceBus.Metrics/Probes/DurationProbe.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using NServiceBus;
+
+class DurationProbe : Probe, IDurationProbe
+{
+    public DurationProbe(string name, string description) : base(name, description)
+    {
+    }
+
+    public void Register(Action<TimeSpan> observer)
+    {
+        observers += observer;
```

Comment:
My assumption that you register to probes forever. I see no value for making them unregisterable/weak. Do you have an example for it @danielmarbach  ?